File: 1877.数组中最大数对和的最小值.py
#
# @lc app=leetcode.cn id=1877 lang=python3
#
# [1877] 数组中最大数对和的最小值
#

# @lc code=start
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        return max(nums[i] + nums[~i] for i in range(len(nums) // 2))
# @lc code=end

File: 1877.数组中最大数对和的最小值.go
/*
 * @lc app=leetcode.cn id=1877 lang=golang
 *
 * [1877] 数组中最大数对和的最小值
 */

// @lc code=start
func minPairSum(nums []int) int {
	sort.Ints(nums)
	n := len(nums)
	ans := 0
	for i := 0; i < n/2; i++ {
		ans = max(ans, nums[i]+nums[n-i-1])
	}
	return ans

}
func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

// @lc code=end

